### PR TITLE
Functions.elementary.complexes [GSoD]

### PR DESCRIPTION
**Pull request by Soumi7**
# Updates functions.elementary.complexes to follow latest docstring style guide

#### References to other Issues or PRs

#19591

#### Brief description of what is fixed or changed
- Added examples and parameters section to docstrings in elementary.complexes 

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/functions/elementary/complexes.py [hunk 1]
```diff
@@ -38,6 +38,18 @@ class re(Function):
     0
     >>> re(im(x) + x*I + 2)
     2
+    >>> re(5 + I + 2)
+    7
+
+    Parameters
+    ==========
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:45]: We would generally add one spacing between paragraph dashes and next like
```suggestion
    ==========

```

File: sympy/functions/elementary/complexes.py [hunk 2]
```diff
@@ -38,6 +38,18 @@ class re(Function):
...
+    exp : expression
+        real, complex or mixed expression
+
+    Returns
+    =======
+    value : int
+        integer value of real part of expression
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:52]: First characters should be capitalized.

File: sympy/functions/elementary/complexes.py (hunk 1 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:45]: Done

File: sympy/functions/elementary/complexes.py (hunk 2 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Got it.

File: sympy/functions/elementary/complexes.py [hunk 3]
```diff
@@ -38,9 +38,24 @@ class re(Function):
...
+    ==========
+
+    exp : expression
+        Real, complex or mixed expression.
+
+    Returns
+    =======
+
+    value : int
+        Integer value of real part of expression.
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/complexes.py:54]: It's not necessarily an integer:
```
In [17]: re((1 + I)/2)
Out[17]: 1/2
```

File: sympy/functions/elementary/complexes.py [hunk 4]
```diff
@@ -654,6 +710,10 @@ class arg(Function):
     pi/2
     >>> arg(sqrt(2) + I*sqrt(2))
     pi/4
+    >>> arg(sqrt(3)/2 + I/2)
+    pi/6
+    >>> arg(4/5 + 3*I/5)
+    0.643501108793284
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:716]: This example can be confusing because it implicitly ignores how `4/5` can turn out to be floating point coefficient than rational.

File: sympy/functions/elementary/complexes.py [hunk 5]
```diff
@@ -92,10 +107,35 @@ def eval(cls, arg):
...
+        >>> re(x+y*I).as_real_imag()
+        (re(x) - im(y), 0)
+
         """
         return (self, S.Zero)
 
     def _eval_derivative(self, x):
+        '''
+        Examples
+        ========
```
Comment (oscarbenjamin) [inline sympy/functions/elementary/complexes.py:131]: We don't need to document this internal function. A demonstration of these capabilities could go in the main class docstring.

File: sympy/functions/elementary/complexes.py (hunk 4 above)
Comment (oscarbenjamin) [inline sympy/functions/elementary/complexes.py:716]: I think this should either use Rational like `(4 + 3*I)/5` or use floats explicitly like `0.8 + 0.6*I`.

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:716]: On the sympy live shell, running this gives atan(3/4). But while running tests, the output is the floating value. Any reason why this is happenning?

File: sympy/functions/elementary/complexes.py (hunk 5 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:131]: Okay

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:131]: So examples of the internal functions will be inside the main class docstring. Noted.

File: sympy/functions/elementary/complexes.py (hunk 4 above)
Comment (sylee957) [reply sympy/functions/elementary/complexes.py:716]: Sympy live can use some hooks to comprehend 3/4 as rationals than floats

File: sympy/functions/elementary/complexes.py (hunk 5 above)
Comment (oscarbenjamin) [reply sympy/functions/elementary/complexes.py:131]: If there is an example for this it should use `diff` rather than `_eval_derivative`. Users shouldn't call this method directly.

File: sympy/functions/elementary/complexes.py [hunk 6]
```diff
@@ -92,10 +107,35 @@ def eval(cls, arg):
...
+        >>> from sympy import re, I
+        >>> from sympy.abc import x, y
+        >>> re(10 + 20*I).as_real_imag()
+        (10, 0)
+        >>> re(10).as_real_imag()
+        (10, 0)
+        >>> re(40*I).as_real_imag()
+        (0, 0)
+        >>> re(x+y*I).as_real_imag()
+        (re(x) - im(y), 0)
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:123]: I'd also find that this example is not intuitive because people may expect this ti give `x` rather than `re(x) - im(y)` in elementary complex algebra setting.
So I'd better state about assumptions `x, y = symbols('x y', real=True)`

File: sympy/functions/elementary/complexes.py (hunk 4 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:716]: I got it. arg(4 + 3*I) gives atan(3/4). arg(4/5 + 3*I/5) is evaluated before being passed, so it returns float.

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:716]: Done.

File: sympy/functions/elementary/complexes.py (hunk 3 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:54]: Changed.

File: sympy/functions/elementary/complexes.py (hunk 6 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:123]: Done.

File: sympy/functions/elementary/complexes.py [hunk 7]
```diff
@@ -140,12 +166,30 @@ class im(Function):
...
+    re(y) + im(x)
+    >>> im(2 + 3*I).as_real_imag()
+    (3, 0)
+    >>> im(21*I + 50).as_real_imag(deep=True)
+    (21, 0)
+
+    Parameters
+    ==========
+
+    exp : expression
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:185]:
```suggestion
    arg : Expr
```
I have seen that if there is doc is available for `Expr` somewhere, it gets collected by some frameworks like napoleon, so we should pinpoint to actual classes (like `Expr`) instead of using notations like (`expression`)

I'd also suggest that the parameter name should match the actual variable name used in `__new__` or (`eval` for sympy functions)

File: sympy/functions/elementary/complexes.py [hunk 8]
```diff
@@ -140,12 +166,30 @@ class im(Function):
...
+        Real, complex or mixed expression.
+
+    Returns
+    =======
+
+    value : int
+        Integer or float value of imaginary part of expression.
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:192]: `re` and `im` works for any kinds of symbolic expression and returns symbolic value for that.
Sot it should better be 
```suggestion
    value : Expr
        Imaginary part of expression.
```

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:192]: Changed.

File: sympy/functions/elementary/complexes.py (hunk 7 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:185]: Yes.

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:185]: Done.

File: sympy/functions/elementary/complexes.py [hunk 9]
```diff
@@ -754,6 +852,35 @@ def _eval_is_algebraic(self):
...
+    >>> transpose(B)
+    B.T
+    >>> transpose(A*B)
+    B.T*A.T
+
+    Parameters
+    ==========
+
+    arg : MatrixSymbol
+        Symbolic representation of a Matrix Object.
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:874]: I think that more survey should be done for this function because it is used in `tensor` or `quantum`.
And it is applied more generally than `MatrixSymbol` at least, e.g. `MatrixExpr`

File: sympy/functions/elementary/complexes.py [hunk 10]
```diff
@@ -754,6 +852,35 @@ def _eval_is_algebraic(self):
...
+
+    Returns
+    =======
+
+    value : MatrixExpr
+        MatrixExpr is the superclass for matric expressions.
+        Represent abstract matrices, linear transformations
+        represented within a particular basis.
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:882]: I don't think that this is a place to explain MatrixExpr here.
Maybe you should explain something like 
`This returns an according transpose object if the object represents any form of linear map. A common example would be a matrix with its usual transpose.`

File: sympy/functions/elementary/complexes.py [hunk 11]
```diff
@@ -838,6 +992,12 @@ class polar_lift(Function):
     >>> polar_lift(4*p)
     4*p
 
+    Parameters
+    ==========
+
+    arg : Expr
+        Real, complex or mixed expression.
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:999]: I don't think that `mixed` is a clear notation because real numbers and real numbers combined with imaginary numbers all belongs to complex numbers. You should better drop that and denote like `Real or complex expression.`

File: sympy/functions/elementary/complexes.py [hunk 12]
```diff
@@ -909,6 +1072,13 @@ class periodic_argument(Function):
     >>> periodic_argument(exp_polar(5*I*pi), pi)
     0
 
+    Parameters
+    ==========
+
+    arg : exp_polar, exp
+        Exp_polar class represents a polar number.
+        Exp represents an exponential expression.
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:1080]: `eval` uses `ar, period` as literal names of arguments, so you should better change to those names to document.

But in my opinion, it is also incorrect that the first type should always be `exp_polar` and the second type is an instance of `Exp`. Both can just be general sympy `Expr, Expr` instance (For example, `periodic_argument(exp(I*pi), 3*pi)` just works and even first is actually neither `exp_polar` nor `exp` but `NegativeOne`)

**Comment by sylee957** [comment]
Often, the documentation changes can be easily done in parallel and I thought that it was sufficient to merge if you haven't touched other functions like `transpose`, `polar_lift` in a05298bcb8677b40e9044a108fd14cee33f47d0e.
For some functions, it would generally require a lot of survey and reviews, so I think that you should drop the WIP status and restrict the scope to begin with a few functions than whole `elementary` module, if there aren't much changes that can be easily copy pasted over.

File: sympy/functions/elementary/complexes.py (hunk 12 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:1080]: I actually documented only the first argument. I am changing the argument names now.

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:1080]: What does Positive real number of infinity mean? In the explanation of principal_argument?

File: sympy/functions/elementary/complexes.py (hunk 9 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:874]: I will look it up and also ask in the gitter room.

File: sympy/functions/elementary/complexes.py (hunk 10 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:882]: Made some changes.

File: sympy/functions/elementary/complexes.py (hunk 11 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:999]: yes.

**Comment by Soumi7** [comment]
I plan to use one PR to make changes to each file in each submodule in functions. That's fine, right?

So this PR will cover all changes made to **complexes.py** in **functions.elementary**.

File: sympy/functions/elementary/complexes.py [hunk 13]
```diff
@@ -909,6 +1080,16 @@ class periodic_argument(Function):
...
 
+    Parameters
+    ==========
+
+    ar : Expr
+        Expression which can represent a exponent or exponent representing a polar
+        number.
+
+    period : Expr
+        Returns value in the interval (-period/2,period/2].
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:1091]: What does `Reaturns value` mean in this?

File: sympy/functions/elementary/complexes.py [hunk 14]
```diff
@@ -1000,6 +1181,15 @@ class principal_branch(Function):
     >>> principal_branch(exp_polar(2*pi*I)*3*z, 2*pi)
     3*principal_branch(z, 2*pi)
 
+    Parameters
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:1184]: I'd also reformat the above examples sections with `Examples` tag

File: sympy/functions/elementary/complexes.py [hunk 15]
```diff
@@ -1000,6 +1181,15 @@ class principal_branch(Function):
...
+    ==========
+
+    x : Expr
+        Expression which can be an exponent or exponent representing a polar
+        number.
+
+    period : Positive real number.
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:1191]: I don't think that these explanations are mathematically correct or informative. and it's just weird to see something like
`exponent or exponent representing a polar number`

I'd just suggest to rewrite these informal statements about parameters
```
    This is a function of two arguments. The first argument is a polar
    number `z`, and the second one a positive real number of infinity, `p`.
    The result is "z mod exp_polar(I*p)".
```
to the format that fits the parameters section, such like
```
x
    A polar number

period
    A positive real number of infinity
```
At least in the words that the original author had used

**Comment by sylee957** [comment]
I think that there are some underlying difficulty for describing the parameters.
However, if you are not sure about how to pick up some accurate mathematical texts for them
I'd just try to choose examples that can show the difference of the choice of the *specific* parameter

For example, you can do something like
```
period : Expr
    >>> periodic_argument(exp_polar(5*I*pi), 2*pi)
    pi
    >>> periodic_argument(exp_polar(5*I*pi), 3*pi)
    -pi
    >>> periodic_argument(exp_polar(5*I*pi), pi)
    0
```
Although there is insufficient text about what is `period`
Putting the examples at the right place can clearly show the behavior sufficiently enough to understand.

File: sympy/functions/elementary/complexes.py (hunk 15 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:1191]: Here we can't mention a certain datatype for the parameters then.

Comment (sylee957) [reply sympy/functions/elementary/complexes.py:1191]: It could be
```
period : Expr
    A positive real number of infinity
```

File: sympy/functions/elementary/complexes.py (hunk 14 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:1184]: Done.

**Comment by Soumi7** [comment]
I put some of the examples for period in the parameters section of periodic_argument()

Screenshot from 2020-09-09 18-10-55
.

File: sympy/functions/elementary/complexes.py [hunk 16]
```diff
@@ -896,18 +1064,32 @@ class periodic_argument(Function):
     logarithm. That is, given a period P, always return a value in
     (-P/2, P/2], by using exp(P*I) == 1.
 
+    Examples
+    ========
+
     >>> from sympy import exp, exp_polar, periodic_argument, unbranched_argument
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:1070]: You may have to delete unused import `periodic_argument` to pass flake8 test

File: sympy/functions/elementary/complexes.py [hunk 17]
```diff
@@ -896,18 +1064,32 @@ class periodic_argument(Function):
...
     >>> from sympy import I, pi
     >>> unbranched_argument(exp(5*I*pi))
     pi
     >>> unbranched_argument(exp_polar(5*I*pi))
     5*pi
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:1075]: All these examples may move to `unbranched_argument` where the documentation is missing.
However, you may have questions how to derive the summary, title there because the doc is completely missing there.

File: sympy/functions/elementary/complexes.py [hunk 18]
```diff
@@ -896,18 +1064,32 @@ class periodic_argument(Function):
...
-    >>> periodic_argument(exp_polar(5*I*pi), 3*pi)
-    -pi
-    >>> periodic_argument(exp_polar(5*I*pi), pi)
-    0
+
+    Parameters
+    ==========
+
+    ar : Expr
+        A polar number.
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:1081]: This can use the examples like below.
I think that you can make make examples that fixes `period` to `2pi` and try out different `exp_polar`.
Although this function can take more general inputs, I think that the standard usage would be using exp_polar with multiples of `pi`

File: sympy/functions/elementary/complexes.py (hunk 16 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:1070]: Yeah. I added another example with periodic argument instead.

File: sympy/functions/elementary/complexes.py (hunk 17 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:1075]: I have updated the doc string. Is it fine?

File: sympy/functions/elementary/complexes.py (hunk 15 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Yes, I have done this now.

File: sympy/functions/elementary/complexes.py [hunk 19]
```diff
@@ -979,6 +1165,26 @@ def _eval_evalf(self, prec):
...
+    Returns periodic argument of arg with period as infinity.
+
+    Examples
+    ========
+
+    >>> from sympy import exp, exp_polar, unbranched_argument
+    >>> from sympy import I, pi
+    >>> unbranched_argument(exp(10*I*pi))
+    0
+    >>> unbranched_argument(exp(25*I*pi))
```
Comment (sylee957) [inline sympy/functions/elementary/complexes.py:1178]: I'd actually try to recommend using `exp_polar` here which shows more intuitive behavior for the users who doesn't know that exp automatically wraps around `2*pi*I`

File: sympy/functions/elementary/complexes.py (hunk 18 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:1085]: Done.

File: sympy/functions/elementary/complexes.py (hunk 19 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Have added two examples with exp_polar and one with exp.

File: sympy/functions/elementary/complexes.py (hunk 17 above)
Comment (sylee957) [reply sympy/functions/elementary/complexes.py:1075]: I think that it looks fine in unbranched_argument

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:1075]: I'll remove those examples from periodic argument then.

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Done.

File: sympy/functions/elementary/complexes.py [hunk 20]
```diff
@@ -140,12 +166,30 @@ class im(Function):
...
+    2
     >>> im(x*I)
     re(x)
     >>> im(re(x) + y)
     im(y)
+    >>> im(x + y*I)
+    re(y) + im(x)
+    >>> im(2 + 3*I).as_real_imag()
+    (3, 0)
+    >>> im(21*I + 50).as_real_imag(deep=True)
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:179]: This doesn't really demonstrate deep=True. I don't see why that needs to be shown here anyway as this is the docstring for `im`.

File: sympy/functions/elementary/complexes.py [hunk 21]
```diff
@@ -140,12 +166,30 @@ class im(Function):
...
-    17
+    >>> im(2*I + 17)
+    2
     >>> im(x*I)
     re(x)
     >>> im(re(x) + y)
     im(y)
+    >>> im(x + y*I)
+    re(y) + im(x)
+    >>> im(2 + 3*I).as_real_imag()
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:177]: What is the purpose of this example?

File: sympy/functions/elementary/complexes.py [hunk 22]
```diff
@@ -422,6 +474,12 @@ class Abs(Function):
     x**2
     >>> abs(-x) # The Python built-in
     Abs(x)
+    >>> Abs(3*x + 2*I)
+    sqrt(9*x**2 + 4)
+    >>> Abs(8*I)
+    8
+    >>> Abs(-x).fdiff()
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:481]: fdiff is not meant to be used by end users.

File: sympy/functions/elementary/complexes.py (hunk 4 above)
Comment (asmeurer) [reply sympy/functions/elementary/complexes.py:726]: Let's avoid examples that use floating point numbers, especially in cases like this where the numbers are not exact.

File: sympy/functions/elementary/complexes.py [hunk 23]
```diff
@@ -710,6 +792,22 @@ class conjugate(Function):
     2
     >>> conjugate(I)
     -I
+    >>> conjugate(3+2*I)
+    3 - 2*I
+    >>> conjugate(5-I)
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:797]: Put spaces around `+` and `-` in expressions, like `conjugate(3 + 2*I)` and `conjugate(5 - I)`.

File: sympy/functions/elementary/complexes.py [hunk 24]
```diff
@@ -754,6 +852,43 @@ def _eval_is_algebraic(self):
...
+    >>> from sympy import Matrix
+    >>> A = MatrixSymbol('A', 25, 9)
+    >>> transpose(A)
+    A.T
+    >>> B = MatrixSymbol('B', 9, 22)
+    >>> transpose(B)
+    B.T
+    >>> transpose(A*B)
+    B.T*A.T
+    >>> Matrix([[4, 5], [2, 1], [90, 12]])
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:870]: Maybe do 

```py
>>> M = Matrix([[4, 5], [2, 1], [90, 12]])
>>> M
>>> transpose(M)
```

to make it clear it is the same matrix on both lines.

File: sympy/functions/elementary/complexes.py [hunk 25]
```diff
@@ -775,6 +910,30 @@ def _eval_transpose(self):
...
+    ==========
+
+    arg : MatrixSymbol
+        Symbolic representation of a Matrix Object.
+
+    Returns
+    =======
+
+    value : MatrixExpr
+        MatrixExpr is the superclass for matric expressions.
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:933]: This doesn't explain what the return value is.

File: sympy/functions/elementary/complexes.py [hunk 26]
```diff
@@ -896,18 +1064,32 @@ class periodic_argument(Function):
...
+
+    Parameters
+    ==========
+
+    ar : Expr
+        A polar number.
+
+    period : Expr
+
+        >>> from sympy import exp_polar, periodic_argument
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:1085]: There shouldn't be examples in the parameters section.

File: sympy/functions/elementary/complexes.py [hunk 27]
```diff
@@ -1000,6 +1206,15 @@ class principal_branch(Function):
...
     3*principal_branch(z, 2*pi)
 
+    Parameters
+    ==========
+
+    x : Expr
+        A polar number.
+
+    period : Expr
+        Positive real number of infinity.
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:1216]:
```suggestion
        Positive real number or infinity.
```

File: sympy/functions/elementary/complexes.py (hunk 26 above)
Comment (asmeurer) [reply sympy/functions/elementary/complexes.py:1085]: I guess this is based on what @sylee957 wrote. But I think it's better to explain here in plain English what the parameter does. The examples section can show what it does in examples. I think it's already explained in the main part of the docstring, so here I would just write something like "the period $P$", which references the mathematical description at the beginning of the docstring.

File: sympy/functions/elementary/complexes.py (hunk 20 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:179]: Removed.

File: sympy/functions/elementary/complexes.py (hunk 21 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:175]: To demonstrate use of as_real_imag()

File: sympy/functions/elementary/complexes.py (hunk 22 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: I see. Removed.

File: sympy/functions/elementary/complexes.py (hunk 23 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Done.

File: sympy/functions/elementary/complexes.py (hunk 25 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py:928]: Added a one line explanation.

File: sympy/functions/elementary/complexes.py (hunk 27 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Changed.

File: sympy/functions/elementary/complexes.py (hunk 26 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Done.

File: sympy/functions/elementary/complexes.py (hunk 24 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Done.

File: sympy/functions/elementary/complexes.py (hunk 27 above)
Comment (sylee957) [reply sympy/functions/elementary/complexes.py]: If this was a typo, then the above summary should be changed too

Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Done.

File: sympy/functions/elementary/complexes.py (hunk 21 above)
Comment (asmeurer) [reply sympy/functions/elementary/complexes.py:175]: But this is the docstring for `im`. `im(x).as_real_imag()` will always return `(0, im(x))`, so there's no reason why anyone would need to call as_real_imag() after using `im`.

File: sympy/functions/elementary/complexes.py [hunk 28]
```diff
@@ -38,9 +38,34 @@ class re(Function):
...
+    7
+    >>> re(10 + 20*I).as_real_imag()
+    (10, 0)
+    >>> re(10).as_real_imag()
+    (10, 0)
+    >>> re(40*I).as_real_imag()
+    (0, 0)
+    >>> re(x+y*I).as_real_imag()
+    (x, 0)
+    >>> re(im(x) + x*I + 2)._eval_derivative(2)
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:51]: Methods that start with an underscore shouldn't be used in doctests.

File: sympy/functions/elementary/complexes.py [hunk 29]
```diff
@@ -38,9 +38,34 @@ class re(Function):
...
     2
+    >>> re(5 + I + 2)
+    7
+    >>> re(10 + 20*I).as_real_imag()
+    (10, 0)
+    >>> re(10).as_real_imag()
+    (10, 0)
+    >>> re(40*I).as_real_imag()
+    (0, 0)
+    >>> re(x+y*I).as_real_imag()
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:49]: I would remove the as_real_imag from the examples here. This should just show what `re` returns. Also, some of these show basically the same thing as the examples above, so one of the two should be removed.

File: sympy/functions/elementary/complexes.py (hunk 4 above)
Comment (asmeurer) [reply sympy/functions/elementary/complexes.py:720]: I think this is still using a float.

File: sympy/functions/elementary/complexes.py [hunk 30]
```diff
@@ -775,6 +905,30 @@ def _eval_transpose(self):
...
+    >>> from sympy.matrices import MatrixSymbol
+    >>> A = MatrixSymbol('A', 10, 5)
+    >>> adjoint(A)
+    Adjoint(A)
+
+    Parameters
+    ==========
+
+    arg : MatrixSymbol
+        Symbolic representation of a Matrix Object.
```
Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:922]: I think arg can be any matrix, not just a MatrixSymbol.

Comment (asmeurer) [inline sympy/functions/elementary/complexes.py:922]:
```suggestion
    arg : Matrix
        Matrix or matrix expression to take the adjoint of.
```

File: sympy/functions/elementary/complexes.py (hunk 25 above)
Comment (asmeurer) [reply sympy/functions/elementary/complexes.py:928]: I think this adjoint can also work on Matrix objects, in which case the result wouldn't be a matrix expression. Probably better to just say `Matrix` for this case. 

Regardless, it doesn't need to explain what `MatrixExpr` is here.

Comment (sylee957) [reply sympy/functions/elementary/complexes.py:928]: Actually, I think that this class was designed to work with more general non-commutative expressions than Matrix and I see some usage of the same operator in quantum module
We may also have to explain what Adjoint is different from adjoint, although I pretty much expect this is from some bad designs

Comment (asmeurer) [reply sympy/functions/elementary/complexes.py:928]: Yes, I think you're right. That explains why it's in this module. We may want to keep this docstring simple for now, and open an issue for figuring out the two things. I've always been a little confused on why these "operator" functions work on plain symbols. I can understand treating `Symbol('x', commutative=False)` as an operator, but even for a default `Symbol('x')` (which is supposed to represent a complex number), both `adjoint(x)` and `Adjoint(x).doit()` don't evaluate.

File: sympy/functions/elementary/complexes.py (hunk 30 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: done.

File: sympy/functions/elementary/complexes.py (hunk 29 above)
Comment (Soumi7) [reply sympy/functions/elementary/complexes.py]: Done.
